hub.py

```diff
@@ -260,6 +260,22 @@ def setup_nfs_share(self):
             f'sudo mkdir -p {self.nfs_share_name} && sudo chown 1000:1000 {self.nfs_share_name} && sudo ls -ld {self.nfs_share_name}'
         ])
 
+    def check_hub_health(self, test_notebook, service_api_token):
+        """
+        After each hub gets deployed, validate that it 'works'.
+
+        Automatically create a temporary user, start their server and run a test notebook, making
+        sure it runs to completion. Stop and delete the test server at the end. Try this steps twice
+        before declaring it a failure. If any of these steps fails, immediately halt further
+        deployments and error out.
+        """
+        hub_url = f'https://{self.spec["domain"]}'
+        # Export the hub health check service as an env var so that jupyterhub_client can read it.
+        os.environ['JUPYTERHUB_API_TOKEN']=service_api_token
```

Comment:
`subprocess.check_call` can take an explicit dictionary of environment variables. To prevent accidental issues caused by setting `os.environ` in our *own* process, let's instead set the environment of the subprocess explicitly?

Reply:
Is this needed by execute_notebook? Ideally this would be passed as a parameter. If it only reads from the environment variable, we should make sure the original value (if it exists) is restored once the call is done, with a `try/finally` block. See https://github.com/yuvipanda/hubploy/blob/master/hubploy/auth.py#L66 for an example.

Reply:
Unfortunately we can't pass the token as a parameter to this function :( (params received by the [`execute_code` function](https://github.com/GeorgianaElena/jupyterhub_client/blob/user-token/jupyterhub_client/execute.py#L45), which is called by e`xecute_notebook` [here](https://github.com/GeorgianaElena/jupyterhub_client/blob/user-token/jupyterhub_client/execute.py#L89))

I followed the example in` hubploy` (thanks <3) and just commited the changes.